subsys/bluetooth/host/keys.c

```diff
@@ -335,17 +364,31 @@ static int keys_set(const char *name, size_t len_rd, settings_read_cb read_cb,
 		BT_ERR("Failed to allocate keys for %s", bt_addr_le_str(&addr));
 		return -ENOMEM;
 	}
+	if (len != BT_KEYS_STORAGE_LEN) {
+#if defined(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+		if (len == (BT_KEYS_STORAGE_LEN_COMPAT)) {
+			BT_WARN("Compatibility loading with no aging counter");
+			memcpy(keys->storage_start, val, len);
+		} else {
+#endif /* CONFIG_BT_KEYS_OVERWRITE_OLDEST */
+			BT_ERR("Invalid key length %zu != %zu", len,
+			       BT_KEYS_STORAGE_LEN);
+			bt_keys_clear(keys);
 
-	if (len_rd != BT_KEYS_STORAGE_LEN) {
-		BT_ERR("Invalid key length %zu != %zu", len,
-		       BT_KEYS_STORAGE_LEN);
-		bt_keys_clear(keys);
-		return -EINVAL;
+			return -EINVAL;
+#if defined(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+		}
+#endif /* CONFIG_BT_KEYS_OVERWRITE_OLDEST */
 	} else {
 		memcpy(keys->storage_start, val, len);
 	}
 
 	BT_DBG("Successfully restored keys for %s", bt_addr_le_str(&addr));
+#if defined(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+	if (aging_counter_val < keys->aging_counter) {
```

Comment:
```suggestion
	if (IS_DEFINED(CONFIG_BT_KEYS_OVERWIRET_OLDEST) &&
	    aging_counter_val < keys->aging_counter) {
```

Reply:
We have no aging counter variable if `CONFIG_BT_KEYS_OVERWIRET_OLDEST` is not defined